### DISCUSSION TITLE
search_full_archive does not support expansions and fields natively

### DISCUSSION THREAD

**Discussion by gabett** [tweepy/tweepy#2054] [Questions] [answered]
Hello,

I am currently performing some search operations by using the _search_full_archive_ method. However, I see the method does not support the _expansions_ parameter as well as the other _{entity}\_fields_ parameters. Indeed, if for example I add the _expansions_ parameter to the method calling, I get the following exception:

`Unexpected parameter: expansions`

What is the usual workaround in this case? I imagine adding the different fields and expansions to the _query_ parameter may be the obvious choice. By doing so however, the maximum number of 128 characters may be easily reached.

**Comment by Harmon758** [comment, marked as answer]
[`API.search_full_archive`](https://tweepy.readthedocs.io/en/v4.12.1/api.html#tweepy.API.search_full_archive) uses Twitter API v1.1, and expansions and fields are a Twitter API v2 paradigm.

Twitter API v1.1, including `API.search_full_archive`, returns fully saturated Status objects.

Otherwise, you can use the Twitter API v2 equivalent method, [`Client.search_all_tweets`](https://tweepy.readthedocs.io/en/v4.12.1/client.html#tweepy.Client.search_all_tweets).
